elastic/kibana
# Add Host Risk metadata data to alerts flyout

## Summary

issue: https://github.com/elastic/security-team/issues/1309

* Add Host Risk metadata data to alerts flyout
* Filter out empty fields from the alerts flyout summary tab
* Update the alerts flyout summary tab "Threat Summary" section to use the same layout as Host Risk. [Read more](https://github.com/elastic/security-team/issues/1764)


<img width="600" alt="Screenshot 2021-09-28 at 17 21 57" src="https://user-images.githubusercontent.com/1490444/135117090-791c2173-20dc-4a03-aced-0d626fd4ee0b.png">

<img width="457" alt="Screenshot 2021-09-29 at 18 30 13" src="https://user-images.githubusercontent.com/1490444/135311350-9c85fb8e-b61a-461c-913a-62463a82bcc6.png">


<img width="457" alt="Screenshot 2021-09-29 at 18 30 13" src="https://user-images.githubusercontent.com/1490444/136229320-c10618f4-9758-446e-a64e-ee8fc40870d0.png">


### How to test Host risk data:
0. Log into the security app and visit the alert pages and open the flyout for an alert. It shouldn't display the host risk data panel.

1. Enable the feature flag and check it again. It should display the panel and show that there is no data for the host. Add `xpack.securitySolution.enableExperimental: ['riskyHostsEnabled']` to `kibana.dev.yml`

2. if you are using a custom `.siem-signals` index as per your `kibana.dev.yml`, run the following reindex query to obtain a `.siem-signals-default-*` index
note: this step is required for properly loading `Drilldown of Host Risk Score` dashboard, as it looks for the index pattern `.siem-signals-default-*`

<details><summary>Create default .siem-signals index</summary>
<p align="center">
<pre><code>
POST _reindex
{
  "source": {
    "index": "my-custom-siem-signals-index-name"
  },
  "dest": {
    "index": ".siem-signals-default-test"
  }
}
</pre></code></details>

3. make the following requests in Kibana Dev Tools
<details><summary>Create ml_host_risk_score_latest index</summary>
<p align="center">
<pre><code>
PUT ml_host_risk_score_latest
{
  "mappings": {
    "properties": {
      "@timestamp": {
        "type": "date"
      },
      "host": {
        "properties": {
          "name": {
            "type": "keyword"
           }
         }
       },
      "ingest_timestamp": {
        "type": "date"
      },
      "risk": {
        "type": "text",
        "fields": {
          "keyword": {
            "type": "keyword",
            "ignore_above": 256
          }
        }
      },
      "risk_score": {
        "type": "float"
      }
    }
  }
}
</code></pre>
</p>
</details>
<details><summary>Add data to ml_host_risk_score_latest index (please replace {MY-HOST-NAME} by a valid hostname)</summary>
<p align="center">
<pre><code>
POST ml_host_risk_score_latest/_doc
{"@timestamp":"2021-02-23T23:00:00.000Z","risk_score":21,"host":{"name":"{MY-HOST-NAME}"},"rules":{"Unusual Linux Username":{"average_risk":21,"rule_count":2,"rule_risk":42}},"ingest_timestamp":"2021-07-27T18:02:08.319296053Z","risk":"Low"}

</code></pre>
</p>
</details>
<details><summary>Create ml_host_risk_score index </summary>
<p align="center">
<pre><code>
POST _reindex
{
  "source": {
    "index": "ml_host_risk_score_latest"
  },
  "dest": {
    "index": "ml_host_risk_score"
  }
}
</pre></code></p></details>

4. Create an index pattern in Kibana / Stack Management / Index Patterns. The name of the id pattern should be `ml-host-risk-score-index-pattern`, and it should point to `ml_host_risk_score`

5. Open the flyout for an event with the same hostname as the document you added to `ml_host_risk_score_latest` {MY-HOST-NAME} 

### How to test Threat Summary new design

1. Install filebeat https://www.elastic.co/guide/en/beats/filebeat/current/filebeat-installation-configuration.html
2. Enable threat intel module `filebeat modules enable threatintel`  https://www.elastic.co/guide/en/beats/filebeat/current/filebeat-module-threatintel.html
3. Start filebeat 
4. Create an `Indicator Match` rule and map one threat intel field.
5. Open the flyout for an generated event.

### Remaining items todo
- [x] Fix rendering order of enriched data and investigation guide
- [x] Change enriched data panel border to make it rounded 
- [x] The URL inside the tooltip isn't clickable

### Checklist

Delete any items that are not applicable to this PR.
